src/InformaticsGateway/ExecutionPlugins/ExternalAppOutgoing.cs

```diff
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FellowOakDicom;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Monai.Deploy.InformaticsGateway.Api;
+using Monai.Deploy.InformaticsGateway.Common;
+using Monai.Deploy.InformaticsGateway.Configuration;
+using Monai.Deploy.InformaticsGateway.Database.Api.Repositories;
+using Monai.Deploy.InformaticsGateway.Logging;
+
+namespace Monai.Deploy.InformaticsGateway.ExecutionPlugins
+{
+    [PluginName("Remote App Execution Outgoing")]
+    public class ExternalAppOutgoing : IOutputDataPlugin
+    {
+        private readonly ILogger<ExternalAppOutgoing> _logger;
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+        private readonly PluginConfiguration _options;
+
+        public string Name => "Remote App Execution Outgoing";
+
+        public ExternalAppOutgoing(
+            ILogger<ExternalAppOutgoing> logger,
+            IServiceScopeFactory serviceScopeFactory,
+            IOptions<PluginConfiguration> configuration)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _serviceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
+            _options = configuration.Value ?? throw new ArgumentNullException(nameof(configuration));
+            if (_options.RemoteAppConfigurations.ContainsKey("ReplaceTags") is false) { throw new ArgumentNullException(nameof(configuration)); }
+        }
+
+        public async Task<(DicomFile dicomFile, ExportRequestDataMessage exportRequestDataMessage)> Execute(DicomFile dicomFile, ExportRequestDataMessage exportRequestDataMessage)
+        {
+            var tags = IDicomToolkit.GetTagArrayFromStringArray(_options.RemoteAppConfigurations["ReplaceTags"]);
+            var outgoingUid = dicomFile.Dataset.GetString(tags[0]);
+
+            var scope = _serviceScopeFactory.CreateScope();
+            var repository = scope.ServiceProvider.GetRequiredService<IRemoteAppExecutionRepository>();
+
+            var existing = await repository.GetAsync(outgoingUid).ConfigureAwait(false);
+
+            if (existing is not null)
+            {
+                PopulateWithStoredProxyValues(dicomFile, tags, existing);
+            }
+            else
+            {
+                var remoteAppExecution = await PopulateWithNewProxyValues(dicomFile, exportRequestDataMessage, tags).ConfigureAwait(false);
+                await repository.AddAsync(remoteAppExecution).ConfigureAwait(false);
+            }
+            return (dicomFile, exportRequestDataMessage);
+        }
+
+        private async Task<RemoteAppExecution> PopulateWithNewProxyValues(DicomFile dicomFile, ExportRequestDataMessage exportRequestDataMessage, DicomTag[] tags)
```

Comment:
When we populate new values, the SeriesInstanceUIDs also need to be the same if they belong to the same Study+Series.  E.g. if you have 4 studies coming in where each includes a random number of series X and each series also contains a random number of SOP instances Y. The, we need to ensure that the outgoing studies have same X number of series and Y number of SOP instances.